cli/lint.rs

```diff
@@ -21,46 +19,50 @@ use deno_lint::rules;
 use deno_lint::rules::LintRule;
 use std::fs;
 use std::path::PathBuf;
-use std::sync::atomic::{AtomicUsize, Ordering};
+use std::sync::atomic::{AtomicBool, Ordering};
 use std::sync::{Arc, Mutex};
 use swc_ecmascript::parser::Syntax;
 
-pub async fn lint_files(args: Vec<String>) -> Result<(), ErrBox> {
+pub async fn lint_files(
+  args: Vec<String>,
+  reporter_kind: LintReporterKind,
+) -> Result<(), ErrBox> {
   let target_files = collect_files(args)?;
   debug!("Found {} files", target_files.len());
 
-  let error_count = Arc::new(AtomicUsize::new(0));
```

Comment:
**Why is this changing to `has_error`?**

Keeping track of the number of error's is now a responsibility of the "reporter". The only thing we need to know is whether or not to `exit(1)` so we just keep track of if we have an error or not.

You'll notice that this fixes a preexisting bug where the lint resulted in a `Err` but we did not `exit(1)` because the the `error_count` was `0`.